src/Common/Exception/EntityNotFound.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Acme\Common\Exception;
+
+interface EntityNotFound extends \Throwable
```

Comment:
Not a big fan of this.

This change comes from the necessity of a handy way to catch an exception from the application side, this means that the application layer is influencing the way our domain is built.
In fact, the only way of communication (and the related influence) should only be from the domain to the boundaries of the application (app/infra).

But if you really like it then it's fine.

Reply:
That's not totally true. There are different cases inside the domain which do you need to differentiate logics between a kind of errors. For example, if there are connection lost error, then you will retry, otherwise, you will interrupt the execution. It's just your domain that gives more information. And with that information, you can do more stuff

Reply:
I don't see any additional info that extension may give you to the domain.
Beside I see that the change has been done for an application layer need.
And if there are retry errors are always inside the implementation, if it goes out it means that is an unhandable one.